reo/scenario.py

```diff
@@ -317,29 +337,108 @@ def setup_pv(pv_dict, latitude, longitude, time_steps_per_hour):
+            try:
+                chp_prime_mover = chp.prime_mover
+            except:
+                chp_prime_mover = None
             absorpchl = AbsorptionChiller(dfm=dfm, max_cooling_load_tons=elecchl.max_cooling_load_tons,
                                           hw_or_steam=boiler.existing_boiler_production_type_steam_or_hw,
-                                          chp_prime_mover=chp.prime_mover,
+                                          chp_prime_mover=chp_prime_mover,
                                           **inputs_dict['Site']['AbsorptionChiller'])
             tmp = dict()
             tmp['installed_cost_us_dollars_per_ton'] = absorpchl.installed_cost_us_dollars_per_ton
             tmp['om_cost_us_dollars_per_ton'] = absorpchl.om_cost_us_dollars_per_ton
             ModelManager.updateModel('AbsorptionChillerModel', tmp, run_uuid)
 
+        # GHP
+        ghp_option_list = []
+        # Call /ghpghx endpoint if only ghpghx_inputs is given, otherwise
+        if inputs_dict["Site"]["GHP"].get("building_sqft") is not None and \
+            inputs_dict["Site"]["GHP"].get("ghpghx_response") in [None, []]:
+            ghpghx_response_list = []
+            # TODO running this twice causes issues? (with no time.sleep(10))
+            if inputs_dict["Site"]["GHP"].get("ghpghx_inputs") in [None, []]:
+                number_of_ghpghx = 1
+                inputs_dict["Site"]["GHP"]["ghpghx_inputs"] = [{}]
+            else:
+                number_of_ghpghx = len(inputs_dict["Site"]["GHP"]["ghpghx_inputs"])
+            for i in range(number_of_ghpghx):
+                ghpghx_post = inputs_dict["Site"]["GHP"]["ghpghx_inputs"][i]
+                ghpghx_post["latitude"] = inputs_dict["Site"]["latitude"]
+                ghpghx_post["longitude"] = inputs_dict["Site"]["longitude"]
+                # Only SpaceHeating portion of Heating Load gets served by GHP, unless allowed by can_serve_dhw
+                if ghpghx_post.get("heating_thermal_load_mmbtu_per_hr") in [None, []]:
+                    if not inputs_dict["Site"]["GHP"].get("can_serve_dhw"):
+                        ghpghx_post["heating_fuel_load_mmbtu_per_hr"] = lpbf_space.load_list
+                    else:
+                        ghpghx_post["heating_fuel_load_mmbtu_per_hr"] = [lpbf_space.load_list[i] + lpbf_dhw.load_list[i] for i in range(len(lpbf_space.load_list))]
+                if dfm.boiler is not None:
+                    ghpghx_post["existing_boiler_efficiency"] = dfm.boiler.boiler_efficiency #boiler.boiler_efficiency
+                else:
+                    ghpghx_post["existing_boiler_efficiency"] = 0.8
+                if ghpghx_post.get("cooling_thermal_load_ton") in [None, []]:
+                    ghpghx_post["cooling_thermal_load_ton"] = [kwt / TONHOUR_TO_KWHT for kwt in dfm.cooling_load.load_list] #lpct.load_list
+                client = TestApiClient()
```

Comment:
@Bill-Becker Why not just use the ghpghx functions directly? And/or call the Julia API? I'm wary of using a testing tool in production.

Reply:
This way leverages all the same code in `/ghpghx/resources.py` for inputs validation, POSTing to julia_api (http.jl /ghpghx endpoint), error handling, and creating and saving the ghpghx db models. This has been in the production environment for several months now with 1000+ runs, and I haven't seen any issues with it. Let me know if you want to discuss alternative solutions.